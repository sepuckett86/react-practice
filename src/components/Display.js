import React from 'react';
import PropTypes from 'prop-types';

function Display({ controls }) {
  return (
    <>
    <p>{controls.title}</p>
    <p>{controls.subtitle}</p>
    </>
  );
}

Display.propTypes = {
  controls: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
  }).isRequired
};

export default Display;
