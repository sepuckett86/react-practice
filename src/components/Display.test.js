import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(
      <Display
        controls={{ title: 'hi', subtitle: 'bye' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
