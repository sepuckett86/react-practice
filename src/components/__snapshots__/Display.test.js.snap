// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<Fragment>
  <p>
    hi
  </p>
  <p>
    bye
  </p>
</Fragment>
`;
